src/foam/box/SocketConnectBox.js

```diff
@@ -37,7 +42,20 @@ foam.CLASS({
             connectTo(this.address).then(function(s) {
               return this.RawSocketBox.create({ socket: s });
             }.bind(this));
-      }
+      },
+      swiftFactory: `
+let fut = Future<FObject>()
```

Comment:
I think for proper multithreading we should move this logic into socketservice, and add synchronizatino there so that multiple socket boxes to the same address receive the same socket.

Reply:
Moved this into a getSocketBoxFuture method on SocketService that will handle not connecting to the same address twice.